### PR TITLE
226 new arena grand crevice - finished initital map

### PR DESCRIPTION
**Pull request by CraZyB1336**
Grand Canyon map.

MainMenuController chooses a random map and continues with this map during the rounds. (In method ChangeScene)

GiantFan / SuperFan - Adds force to player upwards.

DeathBox - Invisible deathbox that eliminates the player.

Whats missing?:
- Model design of the giant fan and superfan (I have the inital model finished just gotta UV-map it...)
- More props.
- Some faces clipping
- Updating the model of the layers. make it more realistic (for instance hand placing each tile. It's a pain, but it looks nicer)

### CODE REVIEW

File: Assets/Scripts/Interactables/DeathTriggerScript.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+using UnityEngine;
+
+public class DeathTriggerScript : MonoBehaviour
+{
+    private void OnTriggerEnter(Collider other)
+    {
+        if (other.TryGetComponent<HitboxController>(out HitboxController hitbox)) 
+        {
+            //hitbox.health.onDeath?.Invoke(null, 9999, new DamageInfo(null, 99999));
```
Comment (toberge) [inline Assets/Scripts/Interactables/DeathTriggerScript.cs:9]: Please remove commented-out code

File: Assets/Scripts/Interactables/GiantFanController.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+public class GiantFanController : MonoBehaviour
+{
+    [SerializeField]
+    private float airForce = 35f;
+
+    private void OnTriggerStay(Collider other)
+    {
+        if (other.gameObject.TryGetComponent<Rigidbody>(out Rigidbody rigidbodyPlayer))
+        {
+            rigidbodyPlayer = other.gameObject.GetComponent<Rigidbody>();
```
Comment (toberge) [inline Assets/Scripts/Interactables/GiantFanController.cs:12]: No need to find this again, you already have the rigidbody right there! `out` parameters are assigned/changed inside the method they're passed to.

File: Assets/Scripts/UI/MainMenuController.cs [hunk 3]
```diff
@@ -24,6 +25,9 @@ public class MainMenuController : MonoBehaviour
     [SerializeField]
     private GalleryMenu galleryMenu;
 
+    [SerializeField]
+    private string[] mapnames;
```
Comment (toberge) [inline Assets/Scripts/UI/MainMenuController.cs:29]: Should be `mapNames`

File: Assets/Scripts/Gamestate/MatchController.cs [hunk 4]
```diff
@@ -154,7 +159,7 @@ public IEnumerator WaitAndStartNextRound()
         yield return new WaitForSeconds(biddingEndDelay);
         // This needs to be called after inputs are set at start the first time this is needed.
         PlayerInputManagerController.Singleton.ChangeInputMaps("FPS");
-        SceneManager.LoadScene("CraterTown");
+        SceneManager.LoadScene(currentMapName);
```
Comment (toberge) [inline Assets/Scripts/Gamestate/MatchController.cs:162]: So we set one map and that's it? No map changes within the round?

**Review by toberge — CHANGES_REQUESTED** [review]
The map looks really good! The fans are a nice addition, even in their placeholder state.

I just have some comments regarding code style, and a question.

File: Assets/Materials/Gate/New Material 1.mat [hunk 5]
```diff
@@ -0,0 +1,133 @@
+%YAML 1.1
```
Comment (Fueredoriku) [inline Assets/Materials/Gate/New Material 1.mat:1]: Not sure what this material is, should be renamed (or maybe deleted if it's not used)

File: Assets/Scripts/Gamestate/MatchController.cs [hunk 6]
```diff
@@ -91,10 +93,13 @@ void Start()
...
         playerFactory = FindObjectOfType<PlayerFactory>();
 
+        if (currentMapName == null)
+            currentMapName = SceneManager.GetActiveScene().name;
+
         // Makes shooting end quickly if testing with 1 player
 #if UNITY_EDITOR
         if (PlayerInputManagerController.Singleton.playerInputs.Count == 1)
-            roundLength = 5f;
+            roundLength = 60f;
```
Comment (Fueredoriku) [inline Assets/Scripts/Gamestate/MatchController.cs:102]: Should be reverted back to 5

File: Assets/Textures/GrandCanyon/rocknormal.png
Comment (Fueredoriku) [inline Assets/Textures/GrandCanyon/rocknormal.png:1]: 📮👀

**Review by Fueredoriku — CHANGES_REQUESTED** [review]
Map looks GREAT! 😁👍

image
It would be nice if you made all the objects like closedGates and spawnpoints children of the "GrandCanyon2" object.
Also, I think we should have 2 more spawns for when there are 4 players.

I also have a couple of nitpicks regarding the lack of naming of some materials ("new material (0, 1, 2 and 3))" and the debug roundtimer length. 
 If you fix those I think it's good to go!

File: Assets/Materials/Gate/New Material 1.mat (hunk 5 above)
Comment (CraZyB1336) [reply Assets/Materials/Gate/New Material 1.mat:1]: deleted material

File: Assets/Textures/GrandCanyon/rocknormal.png
Comment (CraZyB1336) [reply Assets/Textures/GrandCanyon/rocknormal.png:1]: uhm.. yeah

File: Assets/Scripts/Gamestate/MatchController.cs (hunk 4 above)
Comment (CraZyB1336) [reply Assets/Scripts/Gamestate/MatchController.cs:162]: This is the feature after some discussion between me and Fredrik resolved to.

**Review by Fueredoriku — APPROVED** [review]
Looks good! 🎉

**Review by toberge — APPROVED** [review]

**Pull request merged**